### PR TITLE
feat(typography): converting to typescript (#1282)

### LINKED ISSUE #1282: convert @availity/typography package to typescript
### Current Behavior

The `@availity/typography` package is currently in JavaScript, providing limited type checking and autocompletion support.

### Desired Behavior

Migrate the `@availity/typography` package to TypeScript for improved type safety and developer experience.

### Suggested Solution

Convert all JavaScript files in the `@availity/typography` package to TypeScript, following the successful conversion and removal of `d.ts` files.

### Who does this impact? Who is this for?

Everyone utilizing the `@availity/typography` package, particularly those implementing TypeScript in their projects.

### PR DESCRIPTION
**Pull request by clintonlunn**
closes #1282 

having a go at converting a smaller package to typescript

**Before submitting a pull request,** please make sure the following is done:

1. Fork [the repository](https://github.com/availity/availity-react) and create your branch from `master`.
2. Run `yarn` in the repository root.
3. If you've fixed a bug or added code that should be tested, add tests!
4. Ensure the test suite passes (`yarn test`). Tip: `yarn test --watch TestName` is helpful in development.
5. Make sure your code passed the conventional commits check. Read more about [conventional commits](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#summary)

### CODE REVIEW

**Review by nylon22 — DISMISSED** [review]

**Comment by clintonlunn** [comment]
I changed some of the stories to hopefully be more accurate now. I thought the lorem ipsum filler text took away from the readability of the storybook page so i shortened it.

File: packages/typography/src/Agreement.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import React, { ReactNode, ElementType } from 'react';
+import classNames from 'classnames';
+
+interface AgreementProps {
+  /** Custom HTML tag. Can be a string representing an HTML tag or a React component */
+  tag?: ElementType | string;
```
Comment (jordan-a-young) [inline packages/typography/src/Agreement.tsx]: `ElementType` should suffice here. I believe it includes the strings we need so we don't have to specify the `string` type

File: packages/typography/stories/typography.stories.tsx [hunk 2]
```diff
@@ -12,86 +13,40 @@ export default {
...
-      diam. Etiam metus metus, tincidunt nec tellus a, accumsan rhoncus neque. Vivamus nec augue id elit ultricies
-      mollis. Praesent porta metus sed mollis pellentesque. Vivamus interdum tellus ac luctus venenatis. Ut iaculis
-      rhoncus nibh, in elementum elit commodo dictum. Aenean finibus odio elit, a dictum dolor fermentum sit amet.
-    </Disclaimer>
-  </div>
+AgreementStory.storyName = 'Agreement';
+AgreementStory.args = {
+  tag: 'div',
+  className: 'my-agreement-class',
+  children: <p>Sample agreement text</p>,
```
Comment (jordan-a-young) [inline packages/typography/stories/typography.stories.tsx]: I think for this control we should only allow text and move the `<p />` to inside the story already. That way people using the controls only have to worry about the text and not the html tags

File: packages/typography/stories/typography.stories.tsx [hunk 3]
```diff
@@ -12,86 +13,40 @@ export default {
...
+  </Disclaimer>
 );
-_Disclaimer.args = {
+
+DisclaimerStory.storyName = 'Disclaimer';
+DisclaimerStory.args = {
+  tag: 'div',
+  className: 'my-disclaimer-class',
   styled: true,
+  children: <p>Sample disclaimer text</p>,
```
Comment (jordan-a-young) [inline packages/typography/stories/typography.stories.tsx]: Same comment from the agreement args.

File: packages/typography/tests/Agreement.test.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { render, cleanup } from '@testing-library/react';
+import Agreement from '../src/Agreement';
+
+describe('Agreement', () => {
+  afterEach(cleanup);
```
Comment (jordan-a-young) [inline packages/typography/tests/Agreement.test.tsx]: You can remove this cleanup line. The latest version of `@testing-library/react` does this for us

File: packages/typography/tests/Disclaimer.test.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { render, cleanup } from '@testing-library/react';
+import Disclaimer from '../src/Disclaimer';
+
+describe('Disclaimer', () => {
+  afterEach(cleanup);
```
Comment (jordan-a-young) [inline packages/typography/tests/Disclaimer.test.tsx]: Same thing

**Review by jordan-a-young — APPROVED** [review]

**Review by jordan-a-young — CHANGES_REQUESTED** [review]
Misclick. Meant to hit request changes.

**Review by jordan-a-young — APPROVED** [review]

**Pull request merged**